src/Services/ServiceLocator.php

```diff
@@ -0,0 +1,232 @@
+ * @copyright Copyright (c) 2025 OpenCoreEMR Inc
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Services;
+
+use Laminas\ServiceManager\ServiceManager;
+use Laminas\ServiceManager\Config;
+use OpenEMR\Events\Services\ServiceLocatorEvent;
+
+class ServiceLocator implements ServiceLocatorInterface
+{
+    /**
+     * @var ServiceManager|null
+     */
+    private static ?ServiceManager $serviceManager = null;
+
+    /**
+     * @var array|null Cached service configuration
+     */
+    private static $serviceConfig = null;
+
+    /**
+     * Initialize the service manager with configuration
+     *
+     * @param array $config Optional configuration array to merge with loaded config
+     */
+    public static function initialize(array $config = []): void
+    {
+        // Load service configuration from files
+        $loadedConfig = self::loadServiceConfiguration();
+
+        // Merge user config with loaded configuration
+        $mergedConfig = array_merge_recursive($loadedConfig, $config);
+
+        $serviceConfig = new Config($mergedConfig);
+        self::$serviceManager = new ServiceManager();
+        $serviceConfig->configureServiceManager(self::$serviceManager);
+    }
+
+    /**
+     * Load service configuration from configuration files
+     *
+     * @return array Service configuration array
+     */
+    private static function loadServiceConfiguration(): array
+    {
+        if (self::$serviceConfig !== null) {
+            return self::$serviceConfig;
+        }
+
+        // Default empty configuration
+        self::$serviceConfig = [
+            'services' => [],
+            'factories' => [],
+            'invokables' => [],
+            'aliases' => []
+        ];
+
+        // Load main services configuration
+        $configFile = __DIR__ . '/../../config/services.php';
+        if (file_exists($configFile)) {
+            $loadedConfig = require $configFile;
+            if (is_array($loadedConfig)) {
+                self::$serviceConfig = array_merge_recursive(self::$serviceConfig, $loadedConfig);
+            }
+        }
+
+        // Note: Module service replacements should use the event system,
+        // not configuration files. This keeps core services separate from module customizations.
+
+        return self::$serviceConfig;
+    }
+
+    /**
+     * Get a service by name or interface
+     * First checks if modules provide an alternative via event system
+     *
+     * @param string $serviceName The service name or interface
+     * @param array $context Additional context for service creation
+     * @return mixed The service instance
+     * @throws \Laminas\ServiceManager\Exception\ServiceNotFoundException
+     */
+    public static function get(string $serviceName, array $context = [])
+    {
+        // First, dispatch an event to allow modules to provide alternative services
```

Comment:
Is there a way to hit the cache first before hitting the modules?  I haven't dug into the ServiceManager internals so there may be a danger of never hitting the modules altogether by first checking the service manager, just wondering if you thought about this in case any module misbehaves to minimize the damage of a heavy hit on fetching the service instance.